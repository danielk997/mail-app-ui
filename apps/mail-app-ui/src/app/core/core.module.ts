import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { LoginComponent } from './components/login/login.component';
import { HeaderComponent } from './components/header/header.component';
import { SharedModule } from '../shared/shared.module';
import { SideNavComponent } from './components/side-nav/side-nav.component';

@NgModule({
  declarations: [LoginComponent, HeaderComponent, SideNavComponent],
  imports: [BrowserModule, SharedModule],
  exports: [LoginComponent, HeaderComponent, SideNavComponent],
})
export class CoreModule {}
