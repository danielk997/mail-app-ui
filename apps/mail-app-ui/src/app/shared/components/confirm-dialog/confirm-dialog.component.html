<h5>Confirm action</h5>
<hr>
<ng-container [ngSwitch]="data.type">

  <ng-container *ngSwitchCase="ConfirmDialogType.DELETE">
    <p>Are you sure you want to delete this {{data.message ?? ''}}?</p>
  </ng-container>

  <ng-container *ngSwitchCase="ConfirmDialogType.INFO">
    <p>{{data.message ?? ''}}</p>
  </ng-container>

</ng-container>
<hr>
<button mat-flat-button [class]="buttonClassName" (click)="onClick()">{{buttonText}}</button>
<button mat-button [matDialogClose]="null">Cancel</button>
