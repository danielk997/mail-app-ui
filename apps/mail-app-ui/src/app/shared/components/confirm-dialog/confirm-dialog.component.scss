
.delete {
  background-color: red;
  color: white;
}

.info {
  background-color: dodgerblue;
  color: white;
}
