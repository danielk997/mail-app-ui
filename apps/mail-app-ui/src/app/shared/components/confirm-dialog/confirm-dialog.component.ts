import {Component, Inject} from '@angular/core';
import {MAT_DIALOG_DATA} from "@angular/material/dialog";

@Component({
  selector: 'mail-app-ui-confirm-dialog',
  templateUrl: './confirm-dialog.component.html',
  styleUrls: ['./confirm-dialog.component.scss']
})
export class ConfirmDialogComponent {

  ConfirmDialogType = ConfirmDialogType;

  constructor(
    @Inject(MAT_DIALOG_DATA) public data: ConfirmDialogData
  ) {
  }

  onClick() {
    this.data.onClick();
  }

  get buttonClassName(): string {
    switch (this.data.type) {
      case ConfirmDialogType.DELETE:
        return 'delete';
      case ConfirmDialogType.INFO:
        return 'info';
    }
  }

  get buttonText(): string {
    switch (this.data.type) {
      case ConfirmDialogType.DELETE:
        return 'Delete';
      case ConfirmDialogType.INFO:
        return 'Confirm';
    }
  }
}

export interface ConfirmDialogData {
  onClick(): void;

  type: ConfirmDialogType;
  message?: string;
}

export enum ConfirmDialogType {
  DELETE,
  INFO
}




