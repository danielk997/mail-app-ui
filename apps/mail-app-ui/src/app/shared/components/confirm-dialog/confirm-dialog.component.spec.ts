import { ComponentFixture, TestBed } from '@angular/core/testing';

import {ConfirmDialogComponent, ConfirmDialogType} from './confirm-dialog.component';
import {MAT_DIALOG_DATA, MatDialogModule} from "@angular/material/dialog";

describe('ConfirmDialogComponent', () => {
  let component: ConfirmDialogComponent;
  let fixture: ComponentFixture<ConfirmDialogComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [MatDialogModule],
      declarations: [ConfirmDialogComponent],
      providers: [
        {
          provide: MAT_DIALOG_DATA,
          useValue: {}
        }
      ]
    }).compileComponents();

    fixture = TestBed.createComponent(ConfirmDialogComponent);
    component = fixture.componentInstance;
    component.data.type = ConfirmDialogType.INFO;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture.nativeElement.innerHTML).toMatchSnapshot();
  });

  it('should display correct delete dialog', () => {
    const item = 'Item'
    component.data.type = ConfirmDialogType.DELETE;
    component.data.message = item;
    fixture.detectChanges();
    const element: HTMLElement = fixture.nativeElement;
    expect(element.querySelector('p')?.innerHTML).toMatch(`Are you sure you want to delete this ${item}?`);
    expect(element.querySelector('button')?.innerHTML).toMatch(`Delete`);
  });

  it('should display correct info dialog', () => {
    const item = 'Item'
    component.data.type = ConfirmDialogType.INFO;
    component.data.message = item;
    fixture.detectChanges();
    const element: HTMLElement = fixture.nativeElement;
    expect(element.querySelector('p')?.innerHTML).toMatch(`${item}`);
    expect(element.querySelector('button')?.innerHTML).toMatch(`Confirm`);
  });
});
