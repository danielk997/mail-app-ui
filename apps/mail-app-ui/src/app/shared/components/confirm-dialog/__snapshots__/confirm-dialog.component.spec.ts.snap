// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialogComponent should match snapshot 1`] = `
"<h5>Confirm action</h5><hr><!--bindings={
  \\"ng-reflect-ng-switch-case\\": \\"0\\"
}--><p></p><!--ng-container--><!--bindings={
  \\"ng-reflect-ng-switch-case\\": \\"1\\"
}--><!--ng-container--><hr><button mat-flat-button=\\"\\" class=\\"info\\">Confirm</button><button mat-button=\\"\\" type=\\"button\\">Cancel</button>"
`;
